### DISCUSSION TITLE
Two figures on top of each other

### DISCUSSION THREAD

**Discussion by snhansen** [quarto-dev/quarto-cli#6158] [Q&A] [answered]
### Description

I'm trying to stack two figures on top of each other. Here's the code:

	---
	format: html
	---

	```{r}
	#| echo: false
	#| layout-nrow: 2
	#| fig-cap: "Caption."
	#| fig-subcap: 
	#|   - "Subcaption 1."
	#|   - "Subcaption 2."
	#| fig-align: center
	library(ggplot2)
	p1 <- ggplot() +
	  geom_point(data = mtcars, aes(x = wt, y = mpg))

	p1
	p1
	```

However, they're displayed side-by-side. Is that intentional? If so, any way to override this behaviour? I tried with the `layout: [[100], [100]]` option, but that yields the same result. Everything works fine with three or more figures (e.g. two in the top row and one in the bottom row) but not with two figures.

**Comment by cscheid** [comment, marked as answer]
As a workaround, using `layout-ncol: 1` solves the issue for me in your document. I think our logic for `layout-nrow` and `layout-ncol` is "at most n columns or rows", and not "try to do exactly n rows or cols".

We should document this better, and possibly change our algorithm; I agree with you that this is confusing.

**Reply by snhansen** [reply]
Ah, that's good to know. Thanks for the explanation and the workaround.

**Comment by cderv** [comment]
@cscheid it could just be a matter of processing or CSS because it does end up as two rows when a label is added to the chunk 

````markdown
---
format: html
---

```{r}
#| echo: false
#| label: fig-p1
#| layout-nrow: 2
#| fig-cap: "Caption."
#| fig-subcap: 
#|   - "Subcaption 1."
#|   - "Subcaption 2."
#| fig-align: center
library(ggplot2)
p1 <- ggplot() +
  geom_point(data = mtcars, aes(x = wt, y = mpg))

p1
p1
```
````

![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/eca1e737-b7be-4451-a901-e43bc72ba8e5)

Removing the label will lead to two column and one row. 

I guess the two processing should have lead to same layout. Maybe that is what you meant above though.

**Reply by cscheid** [reply]
The label causes the output to be processed as a crossreferenceable figure. (The code paths are not equivalent; this is part of the issue with our figure and crossref codebase.) It'll be easier to fix after I merge the `float-crossref` branch.
